src/System.Security.Cryptography.X509Certificates/tests/CertTests.cs

```diff
@@ -0,0 +1,140 @@
+                                     publicKey, 0,
+                                     pklen);
+
+                Assert.Equal(0x30, publicKey[0]);
+                Assert.Equal(0xB6, publicKey[9]);
+                Assert.Equal(1, publicKey[pklen - 1]);
+            }
+        }
+
+        [Fact]
+        [ActiveIssue(1977, PlatformID.Any)]
+        public static void X509Cert2Test()
+        {
+            const string CertName =
+                @"E=admin@digsigtrust.com, CN=ABA.ECOM Root CA, O=""ABA.ECOM, INC."", L=Washington, S=DC, C=US";
+
+            using (X509Certificate2 cert2 = new X509Certificate2("test.cer"))
+            {
+                Assert.Equal(CertName, cert2.IssuerName.Name);
+                Assert.Equal(CertName, cert2.SubjectName.Name);
+
+                Assert.Equal("ABA.ECOM Root CA", cert2.GetNameInfo(X509NameType.DnsName, true));
+
+                PublicKey pubKey = cert2.PublicKey;
+                Assert.True(pubKey.Key is RSACryptoServiceProvider);
+                Assert.Equal("RSA", pubKey.Oid.FriendlyName);
+
+                Assert.Equal(new DateTime(2009, 7, 9, 10, 33, 53), cert2.NotAfter);
+                Assert.Equal(new DateTime(1999, 7, 12, 10, 33, 53), cert2.NotBefore);
+
+                Assert.Equal("00D01E4090000046520000000100000004", cert2.SerialNumber);
+                Assert.Equal("1.2.840.113549.1.1.5", cert2.SignatureAlgorithm.Value);
+                Assert.Equal("7A74410FB0CD5C972A364B71BF031D88A6510E9E", cert2.Thumbprint);
+                Assert.Equal(3, cert2.Version);
+            }
+        }
+
+        /// <summary>
+        /// This test is for excerising X509Store and X509Chain code without actually installing any certificate 
+        /// </summary>
+        [Fact]
+        public static void X509CertStoreChain()
+        {
+            X509Store store = new X509Store("My", StoreLocation.LocalMachine);
+            store.Open(OpenFlags.ReadOnly);
+            // can't guarantee there is a certificate in store
+            if (store.Certificates.Count > 0)
+            {
+                X509Chain chain = new X509Chain();
+                foreach (X509Certificate2 c in store.Certificates)
+                {
+                    // can't guarantee success, so no Assert 
+                    if (chain.Build(c))
+                    {
+                        foreach (X509ChainElement k in chain.ChainElements)
+                        {
+                            Assert.NotNull(k.Certificate.IssuerName.Name);
+                        }
+                    }
+                }
+            }
+        }
+
+        [Fact]
+        public static void X509Cert2ToStringVerbose()
+        {
+            X509Store store = new X509Store("My", StoreLocation.CurrentUser);
+            store.Open(OpenFlags.ReadOnly);
+
+            foreach (X509Certificate2 c in store.Certificates)
+            {
+                Assert.NotNull(c.ToString(true));
+            }
```

Comment:
Honestly, I've never been sure what the right thing to do with looping over the store is.  "Yes" is probably the correct answer, because X509Store::Close can't (because it doesn't know if you kept a reference).  But I imagine very little customer code Disposes them.